docs/architecture/012-layers.md

```diff
@@ -0,0 +1,81 @@
+# 012: Layers
+
+Authors: @opqdonut @Macroz
+
+This ADR tries to document the current state of the REMS application
+architecture for the backend.
+
+## Layers
+
+### The API Layer
+
+The API layer takes in a HTTP request and transforms it into ideally
+one call of the service layer.
+
+The API layer lives in the `rems.api.` namespaces.
+
+### The Service Layer
+
+Each service represents the public API for a specific part of REMS that is easy to call
+so it must handle any dependencies etc:
+- creating a resource, workflow etc.
+- fetching a complete representation of a resource, workflow etc.
+- business actions like enabling a resource, workflow etc.
+- joining related items like the organization of a resource, ...
+- process manager and async actions of a command
+
+A service will typically call multiple different DB layer namespaces
+to implement the functionality.
+
+The Service layer lives in the `rems.api.services.*` namespaces.
+
+Historical note: the service layer was created in PR #1487 (and
+expanded subsequent PRs like #1490 #1491 #1495 #1854) to avoid
+circular dependencies between db namespaces. That PR contained some
+discussion about what the purpose of the service namespaces should be,
+but no clear conclusion was reached.
+
+### The DB Layer
+
+The purpose of the DB layer is to handle serializing and deserializing
+data into the database. A single namespace in the DB layer should
+handle one concept (for example `rems.db.licenses`). The DB layer can
+also contain simple domain logic (for example `rems.db.attachments`
+has checking for allowed attachment types). Namespaces in the DB layer
+should not depend on each other. Depending on the common
+`rems.db.core` namespace is fine though.
+
```

Comment:
How about adding something about how the db-internal keys should be translated into model keys?

Reply:
I think it should happen here and the keys I mean are mostly old keys or `endt` etc. Also organization is just a string id column and it should be a `{:organization/id xxx}` structure after load?

Reply:
:+1: